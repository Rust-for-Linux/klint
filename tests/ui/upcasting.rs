#![crate_type="lib"]
#![feature(trait_upcasting)]

#[klint::drop_preempt_count(expect = 0)]
trait A {}

#[klint::drop_preempt_count(expect = 1)]
trait B: A {}

fn upcast(x: &dyn B) -> &dyn A {
    x
}
