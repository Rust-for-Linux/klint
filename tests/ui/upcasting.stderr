error: casting between traits with incompatible preemption count properties
  --> $DIR/upcasting.rs:10:5
   |
10 |     x
   |     ^
   |
   = help: adjustment of `dyn B` is 0 and expectation is 1
   = help: while the expected adjustment of `dyn A` is 0 and the expectation is 0

error: aborting due to 1 previous error

